Automattic/jetpack
# Premium Blocks: Handle core/audio block

This PR makes handle the ` core/audio` block upgrade stuff. For this purpose, it implements a refactoring moving some functions, code, etc. to finally remove completely the `extensions/shared/blocks/cover` folder. Everything is handled by premium blocks.

Fixes #14902

#### Changes proposed in this Pull Request:

* Premium Blocks: handle `core/audio` upgrade block


#### Does this pull request change what data or activity we track or use?


no

#### Testing instructions:






* Get sandboxed
* Apply D46955-code and the changes implemented on this PR https://github.com/Automattic/jetpack/pull/16636
* Test in a Simple site with a free plan
* Add a `core/cover` block
* Perform the testing flow for the core/cover block
  - Try up upload a video dropping a video file in the media placeholder area.
  - Try to replace the background uploading a video via the replace control button

* Add a `core/audio` block
  - Try to upload an audio file - confirm that you get the upgrade banner when it tries to upload it.
![image](https://user-images.githubusercontent.com/77539/89355808-b183de00-d692-11ea-9f62-ef7169e0e716.png)

  - Try to upload an unsupported file - confirm you get a generic error.
![image](https://user-images.githubusercontent.com/77539/89355789-a761df80-d692-11ea-8376-03cd4d02f81d.png)


#### Proposed changelog entry for your changes:

* Premium Blocks: handle `core/audio` upgrade block
